Feat: Results pull and Export

thenav56 opened the pull request:
Addresses
- Results pull from firebase
- Exports

## Changes

- **chore: minor fixes**
- **feat(mapping): pull mapping from firebase**
- **feat(exports): add export logic from existing system**

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

thenav56 committed:
chore: minor fixes

- Rename STATS -> EXPORT (continue of existing update 3efbff1f)
- move fd_name and tb_name to utils
- add missing type_enum to CommonAsset
- add get_string_for_filepath for AssetTypeEnum (used later)
- add missing type hints for id
- move meme type map for project export to models.py
- move bot_user to user models
- short the logging output by chaning MainThread -> M

thenav56 committed:
feat(mapping): pull mapping from firebase

thenav56 committed:
feat(exports): add export logic from existing system

tnagorra commented on apps/project/tasks.py:
```diff
@@ -38,3 +38,27 @@ def push_project_to_firebase(project_id: int):
+    if project_id is not None:
+        project = Project.objects.get(pk=project_id)
+    elif project_firebase_id is None:
+        project = Project.objects.get(firebase_id=project_id)
```
Shouldn't we instead be using project_firebase_id instead of project_id here?

tnagorra commented on apps/project/models.py:
```diff
@@ -41,6 +48,30 @@ def get_display(cls, value: typing.Self | int) -> str:
         return "Unknown"
 
+    @staticmethod
+    def get_meme_type(export_type: "ProjectAssetExportTypeEnum"):
```
```suggestion
    @staticmethod
    def get_mime_type(export_type: "ProjectAssetExportTypeEnum"):
```

tnagorra commented on apps/project/exports/tasking_manager_geometries.py:
```diff
@@ -0,0 +1,503 @@
+    project_data = []
+    with gzip.open(gzipped_csv_file, mode="rt") as f:
+        reader = csv.reader(f, delimiter=",")
+        column_index_map = {}
```
Can we add typings here?

tnagorra commented on apps/project/exports/tasking_manager_geometries.py:
```diff
@@ -0,0 +1,503 @@
+        index = column_index_map[label]
+        return modifier(row[index])
+
+    project_data = []
```
Can we add typings here?

tnagorra commented on apps/project/exports/tasking_manager_geometries.py:
```diff
@@ -0,0 +1,503 @@
+    return project_data
+
+
+def yes_maybe_condition_true(x: dict[str, typing.Any]) -> bool:
```
Can we add typings here?

tnagorra committed:
fix(export): add typings

- add typings for export related utilities
- rename meme_type to mimetype
- fix reading project_firebase_id on project export generation
- merge cleanup_temp_tables with temp_results_to_real_table
- rename temp_results_to_real_table to transfer_results_from_temp_tables
- remove reset method from FirebaseCleanup class
- add cleaning up of _root_project in FirebaseCleanup class
- add exec_info=True in logger.error through-out the codebase
- use project_type_enum when generating hot tm geometries
- use firebase_id instead of id when exporting for hot tm geometries
- use logger.warning for expected errors when processing project
